server/mongodb/models/Nonprofit.ts

```diff
@@ -24,6 +27,25 @@ const NonprofitSchema = new Schema<Nonprofit>({
   mission: String,
 });
 
+NonprofitSchema.pre(
+  "remove",
+  async function (next: (err?: CallbackError) => void) {
+    const id = this._id as Types.ObjectId;
+    // Cascade deletes
+    await Promise.all([
+      // delete the user associated with the nonprofit
+      deleteDocumentAndDependencies(UserModel, {
+        nonprofit: id,
+      }),
```

Comment:
same question as before

Reply:
This deletes all users associated with the nonprofit (e.g., the nonprofit field has this id).